File: processes_learning/beam_developer/my_process.erl
-module(my_process).
-export([receive_messages/0]).

receive_messages() ->
  receive
    say_hello ->
      io:fwrite("Hello, I'm the process!");

    say_bye ->
      io:fwrite("Bye, I'm the process!")
  end,
  receive_messages().

%%% How to run this code?
%%% c(my_process).
%%% Pid = spawn(my_process, receive_messages, []).
%%% is_process_alive(Pid).
%%% Pid ! say_hello.
%%% is_process_alive(Pid).
%%% Pid ! say_bye.

File: processes_learning/beam_developer/my_process.ex
defmodule MyProcess do
  @moduledoc"""
    c "my_process.ex"
    pid = spawn(MyProcess, :receive_messages, [])
    send(pid, :say_hello)
    Process.alive?(pid)
  """
  def receive_messages() do
    receive do
      :say_hello->
        IO.puts "Hello, I'm the process!"
      :say_bye ->
        IO.puts "Bye, I'm the process!"
    end
  end
end
